graphql/statistics/fields.ts

```diff
@@ -312,43 +455,167 @@ export class StatisticsResolver {
+                                      ORDER BY "year" ASC, "month" ASC, "status" ASC;`;
+    }
+
+    @FieldResolver((returns) => [ByMonth])
+    @Authorized(Role.ADMIN)
+    async pupilScreenings(@Root() statistics: Statistics, @Arg('status') status: number) {
+        if (status < 0 || status > 3) {
+            throw new PrerequisiteError('Status must be a number between 0 and 3');
+        }
+        return await prisma.$queryRaw`SELECT COUNT(*)::INT                         AS value,
+                                             date_part('year', "createdAt"::date)  AS year,
+                                             date_part('month', "createdAt"::date) AS month
+                                      FROM "pupil_screening"
+                                      WHERE "createdAt" > ${statistics.from}::timestamp
+                                        AND "createdAt" < ${statistics.to}::timestamp
+                                        AND "status" = '${status}'
+                                      GROUP BY "year", "month"
+                                      ORDER BY "year" ASC, "month" ASC;`;
+    }
+
+    @FieldResolver((returns) => Float)
+    @Authorized(Role.ADMIN)
+    async rateSuccessfulCoCs() {
+        const currentDate = new Date(); // Get the current date
+        const previousDate = new Date(currentDate.getFullYear(), currentDate.getMonth() - 8, currentDate.getDate());
+        const successfulScreeningsCount = await prisma.screening.count({
+            where: {
+                success: true,
+                createdAt: { gte: previousDate },
+            },
+            distinct: 'studentId',
+        });
+        const submittedCoCs = await prisma.certificate_of_conduct.count({
+            where: { createdAt: { gte: previousDate } },
+            distinct: 'studentId',
+        });
+        return submittedCoCs / successfulScreeningsCount;
+    }
+
+    @FieldResolver((returns) => Float)
+    @Authorized(Role.ADMIN)
+    async rateTutorFirstMatchStillActiveAfterOneMonth(@Root() statistics: Statistics) {
+        const now = new Date();
+        const oneMonthAgo = new Date(now.getFullYear(), now.getMonth() - 1, now.getDate());
+        const students = await prisma.student.findMany({
+            where: {
+                createdAt: { lte: oneMonthAgo },
+                match: { some: {} },
+            },
+            select: {
+                match: true,
+            },
+        });
+
+        const relevantStudents = students.filter((s) => {
+            const firstMatch = s.match.sort((a, b) => a.createdAt.getTime() - b.createdAt.getTime())[0];
+            return firstMatch.dissolvedAt.getTime() - firstMatch.createdAt.getTime() >= 30 * 24 * 60 * 60 * 1000;
+        });
+
+        return relevantStudents.length / students.length;
+    }
+
+    @FieldResolver((returns) => Float)
+    @Authorized(Role.ADMIN)
+    async rateTargetAudience(@Root() statistics: Statistics) {
```

Comment:
What is "target audience"?

Reply:
It's the pupils who "passed" the pupil screening

Reply:
pupilsInTargetAudienceRate then?